lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1836,84 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    terms.each do |term|
+      matches = filter_sessions_by_search(term)
+      matches.each do |session_id, session|
+        matching_sessions.store(session_id, session)
+      end
+    end
+    matching_sessions
+  end
+
+  def filter_sessions_by_search(search_term)
+    matching_sessions = {}
+    
+    framework.sessions.each do |session_id, session|
+      case search_term.split(":")[0]
+      when "last_checkin"
+        if session.respond_to?(:last_checkin) && session.last_checkin && evaluate_search_criteria(session, search_term)
+          matching_sessions.store(session_id, session)
+        end
+      when "session_type"
+        matching_sessions.store(session_id, session) if evaluate_search_criteria(session, search_term)
+      when "session_id"
+        matching_sessions.store(session_id, session) if evaluate_search_criteria(session, search_term)
+      end
+    end
+    matching_sessions
+  end
+
+  def evaluate_search_criteria(session, search_term)
+    parts = search_term.split(":")
+
+    field = parts[0]
+    operator = parts[1]
+    value = parts[2] if parts[2]
+    return false unless VALID_PARAMS.include? parts[0]
+
+    case field
+    when "last_checkin"
```

Comment:
similar story here to below, if there is an unknown field we should let the user know so they can correct it rather than thinking their search didn't return anything